rcl_logging_log4cxx/src/rcl_logging_log4cxx/rcl_logging_log4cxx.cpp

```diff
@@ -102,12 +102,11 @@ rcl_logging_ret_t rcl_logging_external_initialize(
     // To be compatible with ROS 1, we want to construct a default filename of
     // the form ~/.ros/log/<exe>_<pid>_<milliseconds-since-epoch>.log
 
-    // First get the home directory.
-    const char * homedir = rcutils_get_home_dir();
-    if (homedir == nullptr) {
-      // We couldn't get the home directory; it is not really going to be
-      // possible to do logging properly, so get out of here without setting
-      // up logging.
+    char * logdir = nullptr;
+    rcl_logging_ret_t dir_ret = rcl_logging_get_logging_directory(allocator, &logdir);
+    if (RCL_LOGGING_RET_OK != dir_ret) {
+      // We couldn't get the log directory, so get out of here without setting up
+      // logging.
       return RCL_LOGGING_RET_ERROR;
```

Comment:
```suggestion
      return dir_ret;
```

Reply:
fixed in both implementations in 47bc3e46aa7ba227a0f149f6f8871b92451d25a5